make/targets_list.mk

```diff
@@ -75,6 +75,7 @@ UNSUPPORTED_TARGETS := \
     MIDELICF3 \
     MOTOLAB \
     MULTIFLITEPICO \
+    NUCLEOF303RE \
```

Comment:
One thought I had re this is that now that a large number of targets will become legacy, we should maybe considering separating them into a directory of their own, like `src/main/target_legacy/`. But that is something we can look into once 4.1.0 is out.